src/sqlfluff/rules/L051.py

```diff
@@ -36,23 +43,52 @@ class Rule_L051(BaseRule):
         INNER JOIN baz;
     """
 
+    config_keywords = ["fully_qualify_join_types"]
+
     def _eval(self, context: RuleContext) -> Optional[LintResult]:
-        """INNER JOIN must be fully qualified."""
+        """Fully qualify JOINs."""
+        # Config type hints
+        self.fully_qualify_join_types: str
+
         # We are only interested in JOIN clauses.
-        if context.segment.type != "join_clause":
+        if not context.segment.is_type("join_clause"):
             return None
 
-        # We identify non-lone JOIN by looking at first child segment.
-        if context.segment.segments[0].name != "join":
-            return None
+        join_clause_keywords = [
+            segment for segment in context.segment.segments if segment.type == "keyword"
+        ]
+
+        # We identify LEFT/RIGHT/OUTER JOIN and if the next keyword is JOIN.
+        if (
+            self.fully_qualify_join_types in ["outer", "both"]
+            and join_clause_keywords[0].name in ["right", "left", "full"]
+            and join_clause_keywords[1].name == "join"
+        ):
+            # Insert OUTER after LEFT/RIGHT/OUTER
+            return LintResult(
+                context.segment.segments[0],
+                fixes=[
+                    LintFix.create_after(
+                        context.segment.segments[0],
+                        [WhitespaceSegment(), KeywordSegment("OUTER")],
+                    )
+                ],
+            )
+
+        # We identify lone JOIN by looking at first child segment.
+        if (
+            self.fully_qualify_join_types in ["inner", "both"]
+            and join_clause_keywords[0].name == "join"
+        ):
+            # Replace lone JOIN with INNER JOIN.
+            return LintResult(
+                context.segment.segments[0],
+                fixes=[
+                    LintFix.create_before(
+                        context.segment.segments[0],
+                        [KeywordSegment("INNER"), WhitespaceSegment()],
```

Comment:
Same question about whether we should set the case of the `INNER` based on the case of the other keywords in case someone prefers lowercase?